aws/components/apigateway/state.ftl

```diff
@@ -298,24 +298,28 @@ created in either case.
     [#local customDomainResources = {} ]
     [#list customDomains as domain]
         [#local customFqdn = formatDomainName(customHostName, domain) ]
-        [#local customFqdnParts = splitDomainName(customFqdn) ]
-        [#local customFqdnId = formatResourceId(AWS_APIGATEWAY_DOMAIN_RESOURCE_TYPE, customFqdnParts) ]
-        [#local customDomainResources +=
-            {
-                customFqdn : {
-                    "domain" : {
-                        "Id" : customFqdnId,
-                        "Name" : customFqdn,
-                        "CertificateId" : certificateId,
-                        "Type" : AWS_APIGATEWAY_DOMAIN_RESOURCE_TYPE
-                    },
-                    "basepathmapping" : {
-                        "Id" : formatDependentResourceId(AWS_APIGATEWAY_BASEPATHMAPPING_RESOURCE_TYPE, customFqdnId),
-                        "Stage" : mappingStage,
-                        "Type" : AWS_APIGATEWAY_BASEPATHMAPPING_RESOURCE_TYPE
+        [#if !customFqdn?contains(".") ]
+            [@fatal message="Missing domain name for API gateway" context=solution /]
+        [#else]
```

Comment:
Removing the state will cause issues in the setup routine when it is run. This is still run before an alert is generated. If the state is depdended on in the setup process we won't see this alert.

Reply:
Since the fatal error doesn't have a stop on it and I'm just excluding the customFqdn doesn't it continue to process just without the custom domain; and the state and setup seems to handle a missing customDomains already. What am I missing?